### ISSUE TITLE
Issues decoding unicode values in metadata

### ISSUE THREAD

**Issue by bryannolen** [fireeye/flare-wmi#12]
Not every field is decoding cleanly, horrible hack used to clean as built in python decoding kept dying:

python-cim/cim/objects.py

Line 514, replace function get_string in class DataRegion with:
```
    def get_string(self, ref):
        s = WMIString()
        s.vsParse(self.data, offset=int(ref))
        try:
            return str(s.s.decode("UTF-8"))
        except UnicodeEncodeError:
            if s.s.startswith("Microsoft") and s.s.endswith("Operating System"):
                return("Microsoft Windows Operating System")
            else:
                return("")
```

**Comment by williballenthin** [comment]
hey @bryannolen 

thanks for reporting this issue. is there any chance you can share the underlying data? seems like we could be using the wrong string encoding, but can't be sure without seeing the source string.

**Comment by bryannolen** [comment]
Sadly can't share it because of where I sourced it from, but it is from a very standard Win 7 x64 build.

**Comment by denisegrayson** [comment]
I have seen the same error on every system I have tried recently (5-7 Win 7 x64 systems).  I can't share the source data either, but the full error is "UnicodeEncodeError: 'ascii' codec can't encode character u'\ufffd' in position 26: ordinal not in range(128)", so looks to be some issue with the replacement character.

**Comment by williballenthin** [comment]
i am able to reproduce this issue, and will address it this week.

**Comment by williballenthin** [comment]
failing test case results here: https://travis-ci.org/fireeye/flare-wmi/builds/228059234

notably, the issue is only exercised on py2.7.

**Comment by williballenthin** [comment]
passing test case results here: https://travis-ci.org/fireeye/flare-wmi/builds/228067120

issue is addressed at the library level in commit 7f6a445b3624acdd98571c43c6b6bf434b797a8c. this will be merged, along with a bunch of other enhancements, at the end of this week.

its possible that some tools may still complain about encoding issues, as some of the sample scripts are a little... naive. if you notice this, please do let me know and i'll dev test cases and a patch.

thanks for taking the time to report the issue and provide details!
